src/Settings/LinkAuthWithOs/LinkAuthWithOsScreen.tsx

```diff
@@ -0,0 +1,150 @@
+import {useNavigation} from '@react-navigation/native'
+import React from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {StyleSheet} from 'react-native'
+
+import {useAuthOsAppKey, useAuthOsErrorDecoder, useLoadSecret, useSaveAuthMethod, useSaveSecret} from '../../auth'
+import {Button} from '../../components'
+import globalMessages from '../../i18n/global-messages'
+import {isEmptyString} from '../../legacy/utils'
+import {OsAuthBaseScreen} from '../../OsAuth'
+import {useStorage} from '../../Storage'
+
+export const LinkAuthWithOsScreen = () => {
```

Comment:
`EnableLoginWithOS`?

Reply:
```
const {enable, disable, login} = useLoginWithOS()
```
?

Reply:
`use-Login-` that enable + disable, and login? I'd expect `login` login only.

Reply:
if we consider "login with x" as an entity, then this hook can wrap/provide everything

Reply:
maybe `useOsAuth()`?

Reply:
and `OsAuth` and `PinAuth` are the entities